### PR TITLE
[UWP] Allow embedding Forms page in secondary window

### LINKED ISSUE #2229: [UWP] Embedding Forms page in second UWP Window throws exception
This problem can be reproduced with the Embedding.UWP project on the https://github.com/xamarin/Xamarin.Forms/tree/embedding-uwp-multiwindow branch.

Launching a second UWP window and then attempting to embed a Forms page throws an unhandled exception. The exception occurs in Platform.cs, at

```
_container = new Canvas
{
	Style = (Windows.UI.Xaml.Style)Windows.UI.Xaml.Application.Current.Resources["RootContainerStyle"]
};
```

StackTrace:
```
   at System.Runtime.InteropServices.WindowsRuntime.IMap`2.Lookup(K key)
   at System.Runtime.InteropServices.WindowsRuntime.MapToDictionaryAdapter.Lookup[K,V](IMap`2 _this, K key)
   at System.Runtime.InteropServices.WindowsRuntime.MapToDictionaryAdapter.Indexer_Get[K,V](K key)
   at Xamarin.Forms.Platform.UWP.Platform..ctor(Page page)
   at Xamarin.Forms.Platform.UWP.WindowsPlatform..ctor(Page page)
   at Xamarin.Forms.Platform.UWP.PageExtensions.ToFrameworkElement(VisualElement visualElement)
   at Xamarin.Forms.Platform.UWP.PageExtensions.CreateFrameworkElement(ContentPage contentPage)
   at Xamarin.Forms.Platform.UWP.FormsEmbeddedPageWrapper.OnNavigatedTo(NavigationEventArgs e)
```

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change ###

Adds a check for the existence of the Forms resources when opening a secondary window and loads them if necessary.

Adds a workaround for using the correct dispatcher from Device.BeginInvokeOnMainThread when there are multiple windows present. Also adds a workaround for using multiple Tickers when multiple windows are present.

### Issues Resolved ### 

- fixes #2229
- closes #2432

### API Changes ###
 
Added `protected virtual Ticker GetTickerInstance()` to `Ticker` to allow platforms to modify the number of instances of Tickers available.

### Platforms Affected ### 

- Core
- UWP

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

On UWP, Control Gallery now includes a button to open a secondary window; it will be populated by the test page for issue 2842.

We do not currently have a mechanism for automating tests in a secondary window.

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

File: Xamarin.Forms.Core/Internals/Ticker.cs [hunk 1]
```diff
@@ -42,7 +42,22 @@ protected void OnSystemEnabledChanged()
 		public static Ticker Default
 		{
 			internal set { s_ticker = value; }
-			get { return s_ticker ?? (s_ticker =  Device.PlatformServices.CreateTicker()); }
+			get
+			{
+				if (s_ticker == null)
+				{
+					s_ticker = s_ticker = Device.PlatformServices.CreateTicker();
```
Comment (PureWeen) [inline Xamarin.Forms.Core/Internals/Ticker.cs]:
```suggestion
					s_ticker = Device.PlatformServices.CreateTicker();
```

**Comment by hartez** [comment]
iOS 10 test failure is unrelated to this PR.

**Comment by samhouts** [comment]
@PureWeen You signed off on this, right?

**Comment by wachs** [comment]
Hello everyone,

I did some testing with the solution and found a problem.
I updated the Issue2482.cs class with the following code:
```csharp
var labelRunsBackground = new Label();

layout.Children.Add(labelRunsBackground);

Device.StartTimer(TimeSpan.FromSeconds(1), () => 
{
	Device.BeginInvokeOnMainThread(() => labelRunsBackground.Text = DateTime.Now.ToString("HH:mm:ss"));
	return true;
});
```

In this execution the Wrong Thread problem occurs in the secondary window.

When we worked on the solution (#2432), our main focus was to perform element updates (Labels, Lists, etc.) without the main window selected, so the more complex changes in the framework.

**Pull request merged**